src/prime.ts

```diff
@@ -0,0 +1,99 @@
+export function generatePrime(bitLength: number, options?: { safe: boolean }): bigint {
+    const safe = options?.safe ?? false;
+    if (safe) {
+        bitLength = bitLength - 1;
+    }
+
+    const min = 2n ** BigInt(bitLength - 1);
```

Comment:
here you can generate a random number by sampling random bytes with webcrypto
see: A.5 Converting Random Bits into a Random Number of https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf

so you can remove the `randomBigIntBetween` function